spark/src/main/scala/com/gu/tableversions/spark/filesystem/PathMapper.scala

```diff
@@ -0,0 +1,91 @@
+package com.gu.tableversions.spark.filesystem
+
+import java.net.URI
+
+import com.gu.tableversions.core.{Partition, Version}
+import org.apache.hadoop.fs.Path
+
+/**
+  * Map paths between one "outer" FileSystem (e.g. a ProxyFileSystem) and an underlying one.
+  * For example, by changing the schema, or appending version directories.
+  */
+trait PathMapper {
```

Comment:
I think it'd be good to locate this at `ProxyFileSystem.PathMapper`

Reply:
Agreed.